src/Parser/DemonHunter/Vengeance/Modules/Statistics/SoulFragmentsConsume.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import ExpandableStatisticBox from 'Interface/Others/ExpandableStatisticBox';
+import { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+
+import SoulFragmentsTracker from '../Features/SoulFragmentsTracker';
+
+
+class SoulFragmentsConsume extends Analyzer {
+  static dependencies = {
+    soulFragmentsTracker: SoulFragmentsTracker,
+  };
+
+  castTimestamp = 0;
+  totalSoulsConsumed = 0;
+
+  soulsConsumedBySpell = {};
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SPIRIT_BOMB_TALENT.id && spellId !== SPELLS.SOUL_CLEAVE.id && spellId !== SPELLS.SOUL_BARRIER_TALENT.id) {
+      return;
+    }
+    if (!this.soulsConsumedBySpell[spellId]) {
+      this.soulsConsumedBySpell[spellId] = {name: 0};
+      this.soulsConsumedBySpell[spellId] = {souls: 0};
+      this.soulsConsumedBySpell[1000000] = {souls: 0};
+      this.soulsConsumedBySpell[1000000] = {name: 0};
+      this.soulsConsumedBySpell[1000001] = {souls: 0};
+      this.soulsConsumedBySpell[1000001] = {name: 0};
+    }
+    this.soulsConsumedBySpell[spellId].name = event.ability.name;
+    this.castTimestamp = event.timestamp;
+    this.trackedSpell = spellId;
+  }
+
+  on_byPlayer_removebuffstack(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SOUL_FRAGMENT_STACK.id) {
+      return;
+    }
+    if (event.timestamp - this.castTimestamp < 100) {
+      this.soulsConsumedBySpell[this.trackedSpell].souls += 1;
+      this.totalSoulsConsumed += 1;
+      }
+    }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SOUL_FRAGMENT_STACK.id) {
+      return;
+    }
+    if (event.timestamp - this.castTimestamp < 100) {
```

Comment:
you should be careful doing number comparison on a possibly undefined value (this.castTimestamp), as you can get unexpected results. In this case I'm pretty sure this statement will always evaluate false when this.castTimestamp is undefined, but in other cases you can get in to trouble. Good to start with an undefined check in such statements.

Reply:
i initialize it to 0 for that reason. 
castTimestamp = 0;
So it will evaluate false incase there isnt any cast put into it yet. not good?